projects/ion/src/lib/smart-table/smart-table.component.html

```diff
@@ -99,6 +99,7 @@
               *ngIf="action.confirm"
               ionPopConfirm
               [ionPopConfirmTitle]="action.confirm.title"
+              [ionPopConfirmType]="action.confirm.type"
```

Comment:
You need to add `type` on the interface for confirm action on table.

If `type` is optional, then we will have this problem, as I reported to you on chat:
![image](https://user-images.githubusercontent.com/57760325/234301544-108f24e4-d6ec-42fe-8b5c-7463c57e4e6a.png)

This will happen for this type of assignment to:
![image](https://user-images.githubusercontent.com/57760325/234301780-fa41396c-6a3b-43a4-9960-699948f968a1.png)

I suspect is something in ngOnChanges in AlertComponent, maybe you'll need to check if the change on type is undefined, to set to default value again if it is.